core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -154,6 +150,23 @@ class ReplicaManager(val config: KafkaConfig,
       scheduler.schedule("highwatermark-checkpoint", checkpointHighWatermarks, period = config.replicaHighWatermarkCheckpointIntervalMs, unit = TimeUnit.MILLISECONDS)
   }
 
+  def recordIsrChange(topicAndPartition: TopicAndPartition) {
+    isrChangeSet synchronized {
+      isrChangeSet += topicAndPartition
+    }
+  }
+
+  def maybePropagateIsrChanges() {
+    isrChangeSet synchronized {
+      val now = System.currentTimeMillis()
+      if (isrChangeSet.nonEmpty && now > lastIsrChangeReportMs + config.isrChangePropagateIntervalMs) {
+        ReplicationUtils.propagateIsrChanges(zkClient, isrChangeSet)
```

Comment:
There is no need to check for the time since maybePropagateIsrChanges() is scheduled to run every  config.isrChangePropagateIntervalMs already.